Packs/ShiftLeft/Integrations/shiftleft/shiftleft.py

```diff
@@ -0,0 +1,338 @@
+"""ShiftLeft CORE Integration for Cortex XSOAR (aka Demisto)
+"""
+
+import demistomock as demisto
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import traceback
+from typing import Dict, Any
+
+""" CONSTANTS """
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"  # ISO8601 format with UTC, default in XSOAR
+
+""" CLIENT CLASS """
+
+
+class ShiftLeftClient(BaseClient):
+    """Client class to interact with ShiftLeft V4 api"""
+
+    def get_scopes(self, org_id: str):
+        """Get user scopes"""
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/orgs/{org_id}/my_scopes",
+        )
+
+    def list_apps(
+        self,
+        org_id: str,
+    ) -> Dict[str, str]:
+        """Returns list of apps"""
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/orgs/{org_id}/apps",
+        )
+
+    def list_app_findings(
+        self,
+        org_id: str,
+        app_name: str,
+        severity: Union[str, List[str], None],
+        type: Union[str, List[str], None],
+        version: str,
+    ) -> Dict[str, str]:
+        """Returns list of findings"""
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/orgs/{org_id}/apps/{app_name}/findings",
+            params={"severity": severity, "type": type, "version": version},
+        )
+
+
+def test_module(client: ShiftLeftClient, org_id: str) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ""
+    try:
+        result = client.get_scopes(org_id)
+        message = "ok" if result and result.get("ok") else "Unauthorized"
+    except DemistoException as e:
+        if "Forbidden" in str(e) or "Authorization" in str(e):
+            message = "Authorization Error: make sure access token is correctly set"
+        else:
+            raise e
+    return message
+
+
+def list_apps_command(
+    client: ShiftLeftClient, org_id: str, args: Dict[str, Any]
+) -> CommandResults:
```

Comment:
args here are useless, let's remove them.
```suggestion
def list_apps_command(
    client: ShiftLeftClient, org_id: str) -> CommandResults:
```